identity-appsupport/src/commonMain/kotlin/com/android/identity/appsupport/ui/consent/ConsentModalBottomSheet.kt

```diff
@@ -0,0 +1,465 @@
+                                relyingParty.trustPoint.displayName!!
+                            )
+                        } else if (relyingParty.websiteOrigin != null) {
+                            stringResource(
+                                Res.string.consent_modal_bottom_sheet_share_with_known_requester,
+                                relyingParty.websiteOrigin
+                            )
+                        } else {
+                            stringResource(Res.string.consent_modal_bottom_sheet_share_with_unknown_requester)
+                        },
+                        style = MaterialTheme.typography.bodyMedium,
+                        fontWeight = FontWeight.Bold
+                    )
+                }
+                DataElementGridView(notStoredFields, dontUseColumns)
+            }
+            if (storedFields.size > 0) {
+                if (notStoredFields.size > 0) {
+                    HorizontalDivider(modifier = Modifier.padding(8.dp))
+                }
+                Row(
+                    modifier = Modifier.fillMaxWidth(),
+                    horizontalArrangement = Arrangement.Start
+                ) {
+                    Text(
+                        text = if (relyingParty.trustPoint?.displayName != null) {
+                            stringResource(
+                                Res.string.consent_modal_bottom_sheet_share_and_stored_by_known_requester,
+                                relyingParty.trustPoint.displayName!!
+                            )
+                        } else if (relyingParty.websiteOrigin != null) {
+                            stringResource(
+                                Res.string.consent_modal_bottom_sheet_share_and_stored_by_known_requester,
+                                relyingParty.websiteOrigin
+                            )
+                        } else {
+                            stringResource(Res.string.consent_modal_bottom_sheet_share_and_stored_by_unknown_requester)
+                        },
+                        style = MaterialTheme.typography.bodyMedium,
+                        fontWeight = FontWeight.Bold
+                    )
+                }
+                DataElementGridView(storedFields, dontUseColumns)
+            }
+        }
+    }
+    Spacer(modifier = Modifier.height(2.dp))
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .clip(shape = RoundedCornerShape(0.dp, 0.dp, 16.dp, 16.dp))
+            .background(MaterialTheme.colorScheme.surfaceContainerLowest),
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+        Column(
+            modifier = Modifier.padding(16.dp)
+        ) {
+            // TODO: When we upgrade to a newer version of compose-ui we can use
+            //  AnnotatedString.fromHtml() and clicking the links will also work.
+            //  See b/139326648 for details.
```

Comment:
Do we want the full public link?
https://issuetracker.google.com/issues/139326648

Reply:
Good call.